python/strings_udf/strings_udf/_lib/cudf_jit_udf.pyx

```diff
@@ -22,3 +25,19 @@ def to_string_view_array(Column strings_col):
 
     device_buffer = DeviceBuffer.c_from_unique_ptr(move(c_buffer))
     return Buffer(device_buffer)
+
+
+def from_udf_string_array(DeviceBuffer d_buffer):
+    cdef size_t size = d_buffer.c_size() // 16
```

Comment:
This does not look correct to me. I believe this size is in bytes.
The `column_from_udf_string_array` and the `cpp_free_udf_string_array` expect the `size` parameter to be in the number of elements in the array and not the number of bytes in the device-buffer.
You can compute the number of elements from the bytes using `d_buffer.c_size()/sizeof(udf_string)`

Reply:
yup this is not meant to be hardcoded, will update this.

Reply:
Wait is `//` the same as `/`? I completely ignored the 16 because my brain said that this was a comment.

Reply:
I think this is meant to be basically `d_buffer.c_size() / sizeof(udf_string)`.

Reply:
So then this could be `cdef size_t size = d_buffer.c_size() // sizeof(udf_string)` ?

I'm sorry but '//' is in a read-only part of my brain. Python will just have to pick a different symbol for this operator. How about ÷ ?

Reply:
Yup that is correct - code should already be updated.